packages/platform-browser/src/dom/dom_renderer.ts

```diff
@@ -505,78 +501,105 @@ function isTemplateNode(node: any): node is HTMLTemplateElement {
 }
 
 class ShadowDomRenderer extends DefaultDomRenderer2 {
-  private shadowRoot: any;
+  private shadowRoot: ShadowRoot;
 
   constructor(
     eventManager: EventManager,
-    private hostEl: any,
-    component: RendererType2,
+    private hostEl: HTMLElement,
+    private component: RendererType2,
     doc: Document,
     ngZone: NgZone,
     nonce: string | null,
     platformIsServer: boolean,
     tracingService: TracingService<TracingSnapshot> | null,
-    private sharedStylesHost?: SharedStylesHost,
+    private sharedStylesHost: SharedStylesHost,
+    private styleScopeService: IsolatedStyleScopeService,
   ) {
     super(eventManager, doc, ngZone, platformIsServer, tracingService);
-    this.shadowRoot = (hostEl as any).attachShadow({mode: 'open'});
 
-    // SharedStylesHost is used to add styles to the shadow root by ShadowDom.
-    // This is optional as it is not used by ExperimentalIsolatedShadowDom.
-    if (this.sharedStylesHost) {
+    const isIsolated = component.encapsulation === ViewEncapsulation.ExperimentalIsolatedShadowDom;
+
+    // Only create shadow root in browser environments
+    if (!platformIsServer) {
```

Comment:
to tree shake the error correctly on the client bundle we'll want to use this instead. 
```suggestion
    if (typeof ngServerMode === 'undefined' || !ngServerMode) {
```